sdk/trace/provider.go

```diff
@@ -264,8 +264,9 @@ func WithSpanProcessor(sp SpanProcessor) TracerProviderOption {
 func WithResource(r *resource.Resource) TracerProviderOption {
 	return func(opts *TracerProviderConfig) {
 		if r != nil {
-			opts.resource = r
+			opts.resource = resource.Merge(resource.Environment(), r)
```

Comment:
```suggestion
			opts.resource = resource.Merge(r, resource.Environment())
```

I believe we want the attributes from the environment to take precedence.

Reply:
According to the specification:

> The SDK MUST extract information from the OTEL_RESOURCE_ATTRIBUTES environment variable and merge this, as the secondary resource, with any resource information provided by the user, i.e. the user provided resource information has higher priority.

So I used the Merge function in this manner

Reply:
I asked in the #otel-specification channel and it seems like others interpret it the way you do.  I think this is probably unexpected behavior, but I'm fine with moving forward as it is.